### DISCUSSION TITLE
Wifi usage

### DISCUSSION THREAD

**Discussion by eXqusic92** [doronz88/pymobiledevice3#821] [Q&A]
Is it possible to make this script work via Wi-Fi, not USB cable?

**Comment by wojciech-kulik** [comment]
@doronz88 I noticed that some operations work via Wi-Fi like (un)install and app launch, but it's not possible to create a debug session on the other hand. Is there some kind of limitation?

**Reply by doronz88** [reply]
Never implemented the RSD protocol for wifi connections. Requires research to understand how does XCode perform this operation

**Reply by wojciech-kulik** [reply]
Out of curiosity, how install/uninstall works? Do you recognize wether it's USB or Wi-Fi or does it work out of the box through both channels?

**Reply by doronz88** [reply]
you mean the `apps install/uninstall`? if so, then since the "application-server" is not developer-specific, the communication is done using the old `lockdownd` protocol stack. technically it's just communicating to a predefined tcp-port on device from both usb/wifi.

the `usbmuxd` mac-daemon provides an abstration layer for connecting to any tcp port on each connected device.
for all new developer-specific services i had to implement a whole new protocol-stack which creates a vpn onto device and connects to each service via the new ["RemoteXPC" protocol](https://github.com/doronz88/pymobiledevice3/blob/master/misc/RemoteXPC.md)

**Reply by wojciech-kulik** [reply]
Thank you for the explanation 🍻! That's what I wanted to know.
